Mathlib/RingTheory/MvPolynomial/Symmetric.lean

```diff
@@ -309,4 +312,244 @@ theorem psum_isSymmetric (n : ℕ) : IsSymmetric (psum σ R n) := rename_psum _
 
 end PowerSum
 
+section Exponent
+
+open Classical
+open Composition
+open Finset
+
+variable (σ R) [CommSemiring R] [CommSemiring S] [Fintype σ] [Fintype τ]
+
+@[ext]
+structure Exponent (n : ℕ) :=
+  (exp : σ → ℕ)
+  (sum_eq : ∑ i, exp i = n)
+
+lemma Exponent.single_le_sum {n : ℕ} (s : Exponent σ n) (i : σ) : s.exp i ≤ n := by
+  have h : ∑ i, s.exp i = n := s.sum_eq
+  simp_rw [←h]
```

Comment:
```suggestion
  simp_rw [← h]
```
Are you on an old mathlib? There should be a linter catching this.

Reply:
I will follow the suggestion, ask for access, and create a new branch. These changes are minimal, it's better to switch now. :)